tesla_powerwall/powerwall.py

```diff
@@ -45,149 +46,170 @@ def login_as(
+        return (await self.get_status()).device_type
 
-    def get_serial_numbers(self) -> List[str]:
+    async def get_serial_numbers(self) -> List[str]:
         powerwalls = assert_attribute(
-            self._api.get_powerwalls(), "powerwalls", "powerwalls"
+            await self._api.get_powerwalls(), "powerwalls", "powerwalls"
         )
         return [
             assert_attribute(powerwall, "PackageSerialNumber")
             for powerwall in powerwalls
         ]
 
-    def get_gateway_din(self) -> str:
+    async def get_gateway_din(self) -> str:
         """Return the gateway din."""
-        return assert_attribute(self._api.get_powerwalls(), "gateway_din", "powerwalls")
+        return assert_attribute(
+            await self._api.get_powerwalls(), "gateway_din", "powerwalls"
+        )
 
-    def get_operation_mode(self) -> OperationMode:
+    async def get_operation_mode(self) -> OperationMode:
         operation_mode = assert_attribute(
-            self._api.get_operation(), "real_mode", "operation"
+            await self._api.get_operation(), "real_mode", "operation"
         )
         return OperationMode(operation_mode)
 
-    def get_backup_reserve_percentage(self) -> float:
+    async def get_backup_reserve_percentage(self) -> float:
         return assert_attribute(
-            self._api.get_operation(), "backup_reserve_percent", "operation"
+            await self._api.get_operation(), "backup_reserve_percent", "operation"
         )
 
-    def get_solars(self) -> List[SolarResponse]:
-        return [SolarResponse.from_dict(solar) for solar in self._api.get_solars()]
+    async def get_solars(self) -> List[SolarResponse]:
+        return [
+            SolarResponse.from_dict(solar) for solar in await self._api.get_solars()
+        ]
 
-    def get_vin(self) -> str:
-        return assert_attribute(self._api.get_config(), "vin", "config")
+    async def get_vin(self) -> str:
+        return assert_attribute(await self._api.get_config(), "vin", "config")
 
-    def set_island_mode(self, mode: IslandMode) -> IslandMode:
+    async def set_island_mode(self, mode: IslandMode) -> IslandMode:
         return IslandMode(
             assert_attribute(
-                self._api.post_islanding_mode({"island_mode": mode.value}),
+                await self._api.post_islanding_mode({"island_mode": mode.value}),
                 "island_mode",
             )
         )
 
-    def get_version(self) -> str:
-        version_str = assert_attribute(self._api.get_status(), "version", "status")
+    async def get_version(self) -> str:
+        version_str = assert_attribute(
+            await self._api.get_status(), "version", "status"
+        )
         return version_str.split(" ")[
             0
         ]  # newer versions include a sha trailer '21.44.1 c58c2df3'
 
     def get_api(self) -> API:
         return self._api
 
-    def close(self) -> None:
-        self._api.close()
+    async def close(self):
```

Comment:
```suggestion
    async def close(self) -> None:
```

mypy complained this was untyped

Reply:
Does that mean it exported the typing info for you?   I'm still wondering why it didn't do it for me and I had to add all the cast calls in the HA core.

Reply:
I would assume its seeing the typing based on it warning